acquia/cli
# Use ACLI_ACCESS_TOKEN for auth if present

**Motivation**

Adds support for using access token defined in ACLI_ACCESS_TOKEN.

**Proposed changes**

Before looking for a saved token in cloud_api.conf, ACLI will first check to see if $ACLI_ACCESS_TOKEN is defined as an env var. If so, ACLI assumes that it is a valid access token (not expired) and will use it for authentication.

**Alternatives considered**

Previously considered getting a refresh token from an env var, but that would have required knowing the client id and other inaccessible information.

**Testing steps**


1. Follow the [contribution guide](https://github.com/acquia/cli/blob/master/CONTRIBUTING.md#building-and-testing) to set up your development environment.
2. Clear the kernel cache to pick up new and changed commands: `acli ckc`
3. Get a real access token. It will expire after 5 minutes
* Open Google Chrome Dev Tools
* Change to "network" tab to begin recording requests
* Login to Acquia Cloud
* Immediately, before changing URLs, filter the network tab for the "token" resource and select it
* Select the 'preview' subtab to see find the access token
4. Run `export ACLI_ACCESS_TOKEN=[my token]`
5. If you have a local ~/.acquia/cloud_api.conf, run `mv ~/.acquia/cloud_api.conf ~/.acquia/cloud_api.conf.bak` so that it does not interfere
6. Run an Acquia CLI command and validate that authentication works, even without creds present. E.g., `./bin/acli api:applications:find` 
7. When done, restore your local creds file with `mv ~/.acquia/cloud_api.conf.bak ~/.acquia/cloud_api.conf` 


**Merge requirements**
